### DISCUSSION TITLE
RayTracingPass takes a SceneSrg without "RayTracing"?

### DISCUSSION THREAD

**Discussion by ohmaya** [o3de/o3de#14074] [Help Wanted] [answered]
RayTracingPass.h
- bool m_requiresViewSrg = false;
- bool m_requiresSceneSrg = false; //  **Without RayTracing** ， **RPI::SrgBindingSlot::Scene == 6** 
- bool m_requires**RayTracing**MaterialSrg = false;

RayTracingPass.cpp
- FindShaderResourceGroupLayout(RayTracingGlobalSrgBindingSlot) // 0
- FindShaderResourceGroupLayout(RPI::SrgBindingSlot::Scene) //  **RayTracingSceneSrgBindingSlot == 1 but not used** 
- FindShaderResourceGroupLayout(**RayTracing**MaterialSrgBindingSlot) // 2

**Comment by rgba16f** [comment]
I'm unclear what you are asking here?

**Reply by ohmaya** [reply]
Just compare the middle line of the 2 paragraphs.

**Reply by rgba16f** [reply]
I have compared them and one is a bool and the other is a function. Is there some other attribute you are referring to?

**Reply by ohmaya** [reply]
I learnt from the RTAO sample that **RTSceneSrg and RTMaterialSrg might not be needed**.

**RayTracingPass.h includes a SceneSrg without RT and lacks RTSceneSrg (RTSceneSrg is forced)**, which was strange for me at first.

IIRC, later I found somewhere else that RTSceneSrg is always used. Still need to find if the SceneSrg without RT is redundant. The binding slots are different.

**Comment by ohmaya** [comment, marked as answer]
@kh-huawei helped to improve it? : )
https://github.com/o3de/o3de/commit/faaac41965cd4a3d18c5feac9736915d948b461d
